frame/staking/src/tests.rs

```diff
@@ -2291,10 +2291,11 @@ fn slash_in_old_span_does_not_deselect() {
 			1,
 		);
 
-		// or non-zero.
-		assert_eq!(Staking::force_era(), Forcing::NotForcing);
-		assert!(<Validators<Test>>::contains_key(11));
-		assert!(Session::validators().contains(&11));
+		// the validator doesn't get chilled again
+		assert!(<Staking as Store>::Validators::iter().find(|(stash, _)| *stash == 11).is_some());
+
+		// but we are still forcing a new era
+		assert_eq!(Staking::force_era(), Forcing::ForceNew);
```

Comment:
Please check that this makes sense, I changed the logic so that a validator still gets disabled when there's an offence but it has a new slashing span. It will not get chilled though which was the original behavior.